src/WinRT.Runtime/Projections/Nullable.cs

```diff
@@ -2418,6 +2446,53 @@ unsafe object IWinRTNullableTypeDetails.GetNullableValue(IInspectable inspectabl
                 Marshal.Release(nullablePtr);
             }
         }
+
+        // Delegates are handled separately.
+        Type IWinRTNullableTypeDetails.GetNullableType() => throw new NotImplementedException();
+    }
+
+    public sealed class EnumTypeDetails<T> : IWinRTExposedTypeDetails, IWinRTNullableTypeDetails where T : struct, Enum
+    {
+        [SkipLocalsInit]
+        public ComWrappers.ComInterfaceEntry[] GetExposedInterfaces()
+        {
+            Span<ComWrappers.ComInterfaceEntry> entries = stackalloc ComWrappers.ComInterfaceEntry[2];
+            int count = 0;
+
+            if (FeatureSwitches.EnableIReferenceSupport)
+            {
+                entries[count++] = new ComWrappers.ComInterfaceEntry
+                {
+                    IID = global::WinRT.Interop.IID.IID_IPropertyValue,
+                    Vtable = ABI.Windows.Foundation.ManagedIPropertyValueImpl.AbiToProjectionVftablePtr
+                };
+
+                var type = typeof(T);
+                if (type.IsDefined(typeof(FlagsAttribute)))
+                {
+                    entries[count++] = new ComWrappers.ComInterfaceEntry
+                    {
+                        IID = ABI.System.Nullable_FlagsEnum.GetIID(type),
+                        Vtable = ABI.System.Nullable_FlagsEnum.AbiToProjectionVftablePtr
+                    };
+                }
+                else
+                {
+                    entries[count++] = new ComWrappers.ComInterfaceEntry
+                    {
+                        IID = ABI.System.Nullable_IntEnum.GetIID(type),
+                        Vtable = ABI.System.Nullable_IntEnum.AbiToProjectionVftablePtr
+                    };
+                }
+            }
+
+            return entries.Slice(0, count).ToArray();
+        }
+
+        Type IWinRTNullableTypeDetails.GetNullableType() => typeof(global::System.Nullable<T>);

```

Comment:
Same question here for guarding the generic type instantiation with a feature switch check.